src/main/java/com/openlattice/datastore/search/controllers/SearchController.java

```diff
@@ -476,14 +483,15 @@ public SearchResult executeFQNPropertyTypeSearch( @RequestBody FQNSearchTerm sea
                 neighborList.forEach( neighborEntityDetails -> {
                     neighborsByEntitySet.put( neighborEntityDetails.getAssociationEntitySet().getId(),
                             getEntityKeyId( neighborEntityDetails.getAssociationDetails() ) );
-                    if ( neighborEntityDetails.getNeighborEntitySet().isPresent() ) {
+                    if ( neighborEntityDetails.getNeighborEntitySet().isPresent() && neighborEntityDetails
+                            .getNeighborDetails().isPresent() ) {
                         neighborsByEntitySet.put( neighborEntityDetails.getNeighborEntitySet().get().getId(),
                                 getEntityKeyId( neighborEntityDetails.getNeighborDetails().get() ) );
                     }
                 } )
         );
 
-        List<AuditableEvent> events = new ArrayList<>( neighborsByEntitySet.keySet().size() + 1 );
+        List<AuditableEvent> events = Lists.newArrayListWithExpectedSize( neighborsByEntitySet.keySet().size() + 1 );
```

Comment:
Definite improvement. I'd probably change the sizing to `neighborsByEntitySet.keySet().size() + segments`